### PR TITLE
Allow TestAgent pass a cert and key to request

### PR DESCRIPTION
**Pull request by toefraz**
Superagent added the ability to set the SSL certificate and key parameters for requests, so they need to be added to supertest in order to pass them through to superagent.

### CODE REVIEW

**Comment by mikelax** [comment]
thanks @toefraz can you point me where in the docs superagent is supporting this? I didn't see it on their main doc website, though I could have missed it.

**Comment by toefraz** [comment]
@mikelax it was added in about a month ago with this pull request https://github.com/visionmedia/superagent/pull/1057. It's in 2.3.0

File: lib/agent.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ function TestAgent(app, options) {
   if (!(this instanceof TestAgent)) return new TestAgent(app, options);
   if (typeof app === 'function') app = http.createServer(app);  // eslint-disable-line no-param-reassign
   if (options) this._ca = options.ca;
+  if (options) this._cert = options.cert;
```
Comment (mikelax) [inline lib/agent.js]: cool, do you want to rewrite to clean up slightly? 
```
 if (options) {
    this._ca = options.ca;
    this._key = options.key;
    this._cert = options.cert;
  }
```

Comment (toefraz) [inline lib/agent.js]: No problem. Thanks!

**Comment by toefraz** [comment]
@mikelax, any idea when 2.1.0 might be released?

**Pull request merged**